modules/avformat/avformat.c

```diff
@@ -206,7 +209,24 @@ int avformat_shared_alloc(struct shared **shp, const char *dev)
 	if (err)
 		goto out;
 
-	ret = avformat_open_input(&st->ic, dev, NULL, NULL);
+	input_format = av_find_input_format(dev);
+
+	if (video) {
+		if (size->w) {
+			re_snprintf(buf, sizeof(buf), "%dx%d",
+				    size->w, size->h);
+			info("****** video size = %s\n", buf);
+			av_dict_set(&format_opts, "video_size", buf, 0);
+		}
+		if (fps) {
+			re_snprintf(buf, sizeof(buf), "%d", (int)fps);
+			av_dict_set(&format_opts, "frame_rate", buf, 0);
```

Comment:
the name of the option is "framerate" without the underscore.

the type is `AV_OPT_TYPE_VIDEO_RATE` so I guess both are legal values:

```
29.97
1/30
```

Reply:
After fixing the name of framerate option I made another test.  In config I have
```
video_fps 15
```
and I added some info messages to avformat.c:
```
	if (video && fps) {
		re_snprintf(buf, sizeof(buf), "%2.f", fps);
		ret = av_dict_set(&format_opts, "framerate", buf, 0);
		if (ret != 0) {
			warning("avformat: av_dict_set(framerate) failed"
				" (ret=%s)\n", av_err2str(ret));
			err = ENOENT;
			goto out;
		}
		info("avformat: framerate=%s\n", buf);
		info("avformat: cfg fps=%d\n", (int)cfg->video.fps);
	}
```
When I make a call, I get to log:
```
04-11 09:22:29.744 27531 27585 D Baresip Lib: Set video encoder: H264 packetization-mode=0 (500000 bit/s, 25.00 fps)
04-11 09:22:29.744 27531 27585 D Baresip Lib: avcodec: h264 encoder activated
04-11 09:22:29.744 27531 27585 D Baresip Lib: avcodec: video encoder H264: 25.00 fps, 500000 bit/s, pktsize=1024
04-11 09:22:29.744 27531 27585 D Baresip Lib: video: no video display
04-11 09:22:29.744 27531 27585 D Baresip Lib: avformat: video: alloc dev='android_camera'
04-11 09:22:29.744 27531 27585 D Baresip Lib: avformat: framerate=25.000000
04-11 09:22:29.744 27531 27585 D Baresip Lib: avformat: cfg fps=15
04-11 09:22:29.793 27531 27585 D Baresip Lib: [android_camera @ 0x77e4f44e00] Requested framerate 25 not available, falling back to min: 15 and max: 15 fps
```
So the config setting has no effect when fps is taken from prm->fps instead of cfg.